src/core/utils/Histogram.hpp

```diff
@@ -1,3 +1,5 @@
+#ifndef HISTOGRAM_HPP
```

Comment:
Should correspond to the file path, e.g. CORE_UTILS_HISTOGRAM_HPP, so that there can not be collisions.